frontend/src/components/GoalForm/constants.js

```diff
@@ -4,7 +4,7 @@ import { v4 as uuidv4 } from 'uuid';
 // regex to match a valid url, it must start with http:// or https://, have at least one dot, and not end with a dot or a space
 const VALID_URL_REGEX = /^https?:\/\/.*\.[^ |^.]/;
```

Comment:
```suggestion
const VALID_URL_REGEX = '(?:(?:http|ftp|https|file):\\/\\/)?(?:[\\w%_-]+(?:(?:\\.[\\w%_-]+)+)|(?:\\/[\\w][:]))(?:[\\w\\\\\'\'.,@?^=%&:\\/~+#()-]*[\\w@?^=%&\\/~+#-])';
```

Reply:
We don't want to allow file or ftp. This also looks a little complicated compared to what we actually need according to Angela, which is written out in the comment above. Can you explain why we should make this change?

Reply:
tried adding this locally, seems to have some syntax errors with the slashes in there so I'm going to hold off committing this suggestion for now